src/EnergyPlus/ChillerElectricEIR.cc

```diff
@@ -2662,4 +2677,25 @@ void ElectricEIRChillerSpecs::update(EnergyPlusData &state, Real64 const MyLoad,
     }
 }
 
+bool ElectricEIRChillerSpecs::thermosiphonDisabled(EnergyPlusData &state)
+{
+    if (this->thermosiphonTempCurveIndex > 0) {
+        this->thermosiphonStatus = 0;
+        Real64 dT = this->EvapOutletTemp - this->CondInletTemp;
+        if (dT < this->thermosiphonMinTempDiff) {
+            return true;
+        }
+        Real64 thermosiphonCapFrac = Curve::CurveValue(state, this->thermosiphonTempCurveIndex, dT);
+        Real64 capFrac = this->ChillerPartLoadRatio * this->ChillerCyclingRatio;
+        if (thermosiphonCapFrac >= capFrac) {
+            this->thermosiphonStatus = 1;
+            this->Power = 0.0;
+            return false;
+        }
+        return true;
+    } else {
+        return true;
+    }
+}
+
```

Comment:
There is an issue here with this new function in 4 different places. It would be better to locate this function somewhere so that there would only be 1 new function. And variable names would need to be unified. I should check to see if I made a mistake in Chiller:Electric.

        Real64 capFrac = this->ChillerPartLoadRatio * this->ChillerCyclingRatio; // Electric:EIR
        Real64 capFrac = this->ChillerPartLoadRatio * this->ChillerCyclingRatio; // ReformulatedEIR
        Real64 capFrac = this->CyclingRatio;                                                         // PlantChillers, Chiller:Electric
        Real64 capFrac = this->partLoadRatio * this->cyclingRatio;                     // PlantLoopHeatPump:Cooling:EIR